Developer Samples/CommandLineTools/BuildIrJsRuleApp/Program.cs

```diff
@@ -0,0 +1,210 @@
+            }
+        }
+
+        private static void ShowHelp(OptionSet p)
+        {
+            Console.WriteLine();
+            Console.WriteLine("Usage: BuildIrJsRuleApp.exe [OPTIONS]");
+            Console.WriteLine("Compiles a Rule Application into an irJS executable library using the irDistribution service.");
+            Console.WriteLine();
+            Console.WriteLine("All requests must contain DistributionKey and OutputPath.");
+            Console.WriteLine("   File-based Rule Apps should be specified with RuleAppPath.");
+            Console.WriteLine("   Catalog-based Rule Apps should be specified with CatalogUri, CatalogUsername, CatalogPassword, and CatalogRuleAppName (optionally CatalogRuleAppLabel).");
+            Console.WriteLine();
+            Console.WriteLine("Options:");
+            p.WriteOptionDescriptions(Console.Out);
+            Console.WriteLine();
+        }
+        private static void RetrieveAndWriteIrJSFromDistributionService(RuleApplicationReference ruleRef, string distroKey, string outputPath)
+        {
+            Console.WriteLine();
+
+            Console.WriteLine("Requesting compiled JS library from Distribution Service...");
+            var js = CallDistributionServiceAsync(ruleRef, "https://api.distribution.inrule.com/", distroKey).Result;
+
+            if (!string.IsNullOrEmpty(js))
+            {
+                Console.WriteLine("Received compiled library, writing out to " + outputPath);
+                try
+                {
+                    File.WriteAllText(outputPath, js);
+                    Console.WriteLine("Compiled and wrote out Javascript Rule App");
+                    Console.WriteLine();
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine("Error writing out compiled JavaScript file: " + ex.Message);
+                }
+            }
+        }
+        public static async Task<string> CallDistributionServiceAsync(RuleApplicationReference ruleApplicationRef, string serviceUri, string subscriptionKey)
+        {
+            using (var client = new HttpClient())
+            using (var requestContent = new MultipartFormDataContent())
+            {
+                HttpResponseMessage result = null;
+                try
+                {
+                    client.BaseAddress = new Uri(serviceUri);
+
+                    // Build up our request by reading in the rule application
+                    var ruleApplication = ruleApplicationRef.GetRuleApplicationDef();
+                    var httpContent = new ByteArrayContent(Encoding.UTF8.GetBytes(ruleApplication.GetXml()));
+                    requestContent.Add(httpContent, "ruleApplication", ruleApplication.Name + ".ruleapp");
+
+                    // Tell the server we are sending form data
+                    client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("multipart/form-data"));
```

Comment:
Might want to also add the appropriate HTTP header (I forget the exact one OTOMH) for telling the server to keep the connection alive while waiting for the service to respond